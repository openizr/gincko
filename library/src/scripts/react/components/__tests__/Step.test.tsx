/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { act } from 'react-dom/test-utils';
import Step from 'scripts/react/components/Step';
import { render, unmountComponentAtNode } from 'react-dom';

type Any = any; // eslint-disable-line @typescript-eslint/no-explicit-any

jest.mock('scripts/react/components/Field', () => ({ onUserAction }: Any): JSX.Element => {
  onUserAction();
  return <div id="Field" />;
});
jest.mock('sonar-ui/react', () => ({
  markdown: (value: string): string => value,
  buildClass: (...values: string[]): string => values.join(' '),
}));

describe('react/components/Step', () => {
  let container = document.createElement('div');
  const onUserAction = jest.fn();
  const customComponents = {};

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('active step', () => {
    act(() => {
      render(<Step
        id="step"
        status="success"
        isActive
        index={0}
        fields={[{
          id: 'Message',
          type: 'Unknown',
          status: 'initial',
          options: {},
        }]}
        customComponents={customComponents}
        onUserAction={onUserAction}
      />, container);
    });
    expect(container).toMatchSnapshot();
    expect(onUserAction).toHaveBeenCalledTimes(1);
  });

  test('inactive step', () => {
    act(() => {
      render(<Step
        id="step"
        status="success"
        isActive={false}
        index={0}
        fields={[{
          id: 'Message',
          type: 'Unknown',
          status: 'initial',
          options: {},
        }]}
        customComponents={customComponents}
        onUserAction={onUserAction}
      />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
