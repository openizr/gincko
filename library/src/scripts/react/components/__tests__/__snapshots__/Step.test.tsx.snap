// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/components/Step active step 1`] = `
<div>
  <div
    class="gincko__step success,step,active"
  >
    <div
      class="gincko__step__fields"
    >
      <div
        id="Field"
      />
    </div>
  </div>
</div>
`;

exports[`react/components/Step inactive step 1`] = `
<div>
  <div
    class="gincko__step success,step,"
  >
    <div
      class="gincko__step__fields"
    >
      <div
        id="Field"
      />
    </div>
  </div>
</div>
`;
